lexer.py

```diff
@@ -0,0 +1,127 @@
+from tokens import TokenTypes, Token
+
+###############################################################################
+#                                                                             #
+#   LEXER                                                                     #
+#                                                                             #
+###############################################################################
+
+ESCCHARS = {
+    "n":  "\n",
+    "t":  "\t",
+    "\\": "\\",
+    "'": "'",
+    "r":  "\r",
+}
+
+RESERVED_WORDS = {
+            "SELECT",
+            "FROM",
+            "EDGES",
+            "NODES",
+            "TRAVERSE",
+            "BY",
+            "DEPTH",
+            "BREADTH",
+            "WHERE",
+            "LIMIT",
+}
+
+class Lexer(object):
```

Comment:
You don't need to specify the parent if the parent is `object`, `class Lexer:` will do fine.